core/negotiation.txt

```diff
@@ -0,0 +1,78 @@
+# the conftest1 module is always accepted
+(want core1)
+(have core1.X)
+(want conftest1)
+(have conftest1.3)
+---
+(want core1 conftest1)
+(have core1.X conftest1.3)
+---
+
+# the conftest2 module is never accepted
+(want core1)
+(have core1.X)
+(want conftest2)
+(have)
+---
+(want core1 conftest1 conftest2)
+(have core1.X conftest1.3)
+---
+(want core1 conftest2 conftest1)
+(have core1.X conftest1.3)
+---
+
+# two obviously named capabilities
+(want core1 conftest1 conftest1.acceptable conftest1.unacceptable)
+(have core1.X conftest1.3 conftest1.acceptable)
+---
+
+# capability cannot be accepted without its module
+(want core1 conftest1.acceptable conftest1.unacceptable)
+(have core1.X)
+---
+
+# cannot agree to a capability before having agreed to the module
+(want core1 conftest1.acceptable conftest1.unacceptable conftest1)
+(have)
+---
+(want core1)
+(have core1.X)
+(want conftest1.acceptable conftest1.unacceptable conftest1)
+(core1.nope)
+---
+
+# everything must be barewords in `want` and `have` messages
+("want" core1)
+(have)
```

Comment:
A server must return (nope), if "it cannot determine the message type of the invalid message".
Is it guaranteed that every spec-obedient server can determine the message type `want` from the quoted string "want"? Is it already defined what determing a message type entails exactly?